### PR TITLE
architect ;p

### PR DESCRIPTION
**Pull request by geriamir**
Moved suitcase detection to different class.
We need to test it on Hololens before merge!

### CODE REVIEW

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Terminaltor.Detection;
+
+namespace AssemblyCSharpWSA
+{
+    public class Flight
+    {
+        private readonly IValidationStrategy _validationStrategy;
```
Comment (ashibud) [inline CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: didn't we say we let the I thing go away?

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        private readonly List<SuitcaseDetection> _suitcaseDetections = new List<SuitcaseDetection>();
+
+        public ReadOnlyCollection<SuitcaseDetection> Suitcases => new ReadOnlyCollection<SuitcaseDetection>(_suitcaseDetections);
```
Comment (ashibud) [inline CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: I think this should be an actual suitcase, not a detection

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/SuitcaseDetector.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    if (prediction.probability < probabilityThreshold)
+                    {
+                        Debug.Log("Already at the threshold. Stopping");
+                        break;
+                    }
+
+                    if (prediction.tagName != suitcaseLabel)
+                    {
+                        Debug.Log("Skipping non-suitcase: " + prediction.tagName);
+                        break;
```
Comment (ashibud) [inline CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/SuitcaseDetector.cs]: continue (see my open PR, please merge it to your branch

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/ValidationStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        private readonly int _maxSuitcases;
+
+        public ValidationStrategy(int maxSuitcases)
+        {
+            _maxSuitcases = maxSuitcases;
+        }
+
+        public bool CanAddSuitcase(Flight flight, SuitcaseDetection suitcaseDetection)
+        {
+            return flight.Suitcases.Count != _maxSuitcases;
```
Comment (ashibud) [inline CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/ValidationStrategy.cs]: Use "<" (less than) - as it shows better intent

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs (hunk 1 above)
Comment (geriamir) [reply CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: Lol, didn't even notice, it was done by ReSharper

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs (hunk 2 above)
Comment (geriamir) [reply CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: Yes, you are correct, but right now I wanted to keep the same elements as the prediction itself.
If the change works I will make further fixes.

File: CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs (hunk 1 above)
Comment (ashibud) [reply CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: So also consider (not using) the "_" (underscore) as a member prefix

Comment (geriamir) [reply CustomVisionObjDetection/App/GeneratedProjects/UWP/Assembly-CSharp/Flight.cs]: Also ReSharper :\

**Pull request merged**